GoogleTagManagerBridge/GoogleTagManagerBridge/ACCCustomTagProvider.m

```diff
@@ -0,0 +1,454 @@
+/*!
+ * @file ACCCustomTagProvider.h
+ * @author Accengage
+ * @copyright  © 2010 - present Accengage, Inc. All rights reserved.
+ */
+
+
+#import "ACCCustomTagProvider.h"
+#import "Accengage/Accengage.h"
+
+/* Accengage Actions */
+NSString *const KEY_ACTION = @"acc_action";
+NSString *const ACTION_TRACK_EVENT = @"track_event";
+NSString *const ACTION_TRACK_LEAD = @"track_lead";
+NSString *const ACTION_TRACK_ADD_TO_CART = @"track_add_to_cart";
+NSString *const ACTION_TRACK_PURCHASE = @"track_purchase";
+NSString *const ACTION_SET_UDI = @"set_udi";
+NSString *const ACTION_DELETE_UDI = @"delete_udi";
+NSString *const ACTION_INCREMENT_UDI = @"increment_udi";
+NSString *const ACTION_DECREMENT_UDI = @"decrement_udi";
+
+/* Track Event */
+NSString *const KEY_EVENT_ID = @"acc_event_id";
+
+/* Track Lead */
+NSString *const KEY_LEAD_LABEL = @"acc_lead_label";
+NSString *const KEY_LEAD_VALUE = @"acc_lead_value";
+
+/* Track Add To Cart */
+NSString *const KEY_CART_ID = @"acc_cart_id";
+NSString *const KEY_ITEM_ID = @"acc_item_id";
+NSString *const KEY_ITEM_NAME = @"acc_item_label";
+NSString *const KEY_ITEM_CATEGORY = @"acc_item_category";
+NSString *const KEY_ITEM_CURRENCY = @"acc_item_currency";
+NSString *const KEY_ITEM_PRICE = @"acc_item_price";
+NSString *const KEY_ITEM_QUANTITY = @"acc_item_quantity";
+NSString *const KEY_ITEM_BRAND = @"acc_item_brand";
+
+/* Track Purchase */
+NSString *const KEY_PURCHASE_ID = @"acc_purchase_id";
+NSString *const KEY_PURCHASE_CURRENCY = @"acc_purchase_currency";
+NSString *const KEY_PURCHASE_TOTAL_PRICE = @"acc_purchase_total_price";
+NSString *const KEY_PURCHASE_ITEMS = @"acc_purchase_items";
+
+/* Accengage date format */
+NSString *const DATE_FORMAT = @"yyyy-MM-dd HH:mm:ss zzz";
+
+@implementation ACCCustomTagProvider
+
+///--------------------------------------
+#pragma mark - TAGCustomFunction delegate -
+///--------------------------------------
+
+- (NSObject*)executeWithParameters:(NSDictionary*)parameters {
+    
+    if (!parameters) {
+        NSLog(@"ACCCustomTagProvider | No parameters found");
+        return nil;
+    }
+    
+    NSString *acccengageAction = parameters[KEY_ACTION];
```

Comment:
accengage
Also, check typos like this in all classes...

Reply:
Done